lib/spack/llnl/util/lock.py

```diff
@@ -267,24 +340,65 @@ def acquire_write(self, timeout=None):
         """
         timeout = timeout or self.default_timeout
 
+        locktype = 'WRITE LOCK'
         if self._writes == 0:
-            self._debug(
-                'WRITE LOCK: {0.path}[{0._start}:{0._length}] [Acquiring]'
-                .format(self))
+            self._log_acquiring(locktype)
             # can raise LockError.
             wait_time, nattempts = self._lock(fcntl.LOCK_EX, timeout=timeout)
-            self._acquired_debug('WRITE LOCK', wait_time, nattempts)
             self._writes += 1
+            self._log_acquired(locktype, wait_time, nattempts)
 
             # return True only if we weren't nested in a read lock.
             # TODO: we may need to return two values: whether we got
             # the write lock, and whether this is acquiring a read OR
             # write lock for the first time. Now it returns the latter.
             return self._reads == 0
         else:
+            # Nothing is done with the provided lock timeout so log a warning
+            # if it is different from the one being used.
+            self._log_timeout_warning(timeout)
+
             self._writes += 1
             return False
 
+    def downgrade_write(self, timeout=None):
```

Comment:
I find the name here a little confusing because I have to remember that `write` is what it's downgrading *from*.  Can we change this to `write_to_read` or `downgrade_write_to_read`?

Reply:
Sure.  Picked the latter.